packages/zksend/src/experimental/wallet/index.ts

```
@@ -0,0 +1,468 @@
+				version: '1.0.0',
+				connect: this.#connect,
+			},
+			'standard:disconnect': {
+				version: '1.0.0',
+				disconnect: this.#disconnect,
+			},
+			'standard:events': {
+				version: '1.0.0',
+				on: this.#on,
+			},
+			'sui:signTransactionBlock': {
+				version: '1.0.0',
+				signTransactionBlock: this.#signTransactionBlock,
+			},
+			'sui:signTransaction': {
+				version: '2.0.0',
+				signTransaction: this.#signTransaction,
+			},
+			'sui:signPersonalMessage': {
+				version: '1.1.0',
+				signPersonalMessage: this.#signPersonalMessage,
+			},
+			'sui:signAndExecuteTransaction': {
+				version: '2.0.0',
+				signAndExecuteTransaction: this.#signAndExecuteTransaction,
+			},
+		};
+	}
+
+	constructor({
+		name,
+		origin = DEFAULT_STASHED_ORIGIN,
+	}: {
+		name: string;
+		origin?: string;
+		chain?: SuiChain;
+	}) {
+		this.#accounts = [];
+		this.#events = mitt();
+		this.#origin = origin;
+		this.#name = name;
+		this.#embeddedIframe = null;
+		this.#walletStatusCheckEnabled = false;
+		this.#walletStatusIntervalId = null;
+	}
+
+	#handleWalletStatusMessage = (event: MessageEvent) => {
+		if (event.origin !== stashedWalletOrigin || !this.#walletStatusCheckEnabled) return;
+		try {
+			const message = parse(IframeMessageWalletStatusPayload, event.data);
+			if (message.type === 'WALLET_STATUS') {
+				this.#accounts.forEach((account) => {
+					const foundAddress = message.payload.accounts.some(
+						(item) => item.account.address === account.address,
+					);
+					if (!foundAddress) {
+						this.removeAccount(account.address);
+					}
+				});
+			}
+		} catch (error) {
+			console.warn('Invalid wallet status message:', error);
+		}
+	};
+
+	// Function to clear the interval
+	#clearWalletStatusInterval = () => {
+		if (this.#walletStatusIntervalId !== null) {
+			clearInterval(this.#walletStatusIntervalId);
+			this.#walletStatusIntervalId = null;
+		}
+	};
+
+	#embedStashedIframe = () => {
+		try {
+			/* @ts-ignore */
```

Use expect-error instead and document why you're ignoring the error